src/test/kotlin/uk/gov/justice/hmpps/sqs/SqsQueueAdminServiceTest.kt

```diff
@@ -0,0 +1,240 @@
+            assertThat(it.queueUrl).isEqualTo("dlqUrl")
+            assertThat(it.maxNumberOfMessages).isEqualTo(1)
+          }
+        )
+      }
+
+      @Test
+      fun `Deletes message from the dlq`() {
+        queueAdminService.transferAllMessages(TransferMessagesRequest(dlqSqs, "dlqUrl", queueSqs, "queueUrl"))
+
+        verify(dlqSqs).deleteMessage(
+          check {
+            assertThat(it.queueUrl).isEqualTo("dlqUrl")
+            assertThat(it.receiptHandle).isEqualTo("message-receipt-handle")
+          }
+        )
+      }
+
+      @Test
+      fun `Sends message to the main queue`() {
+        queueAdminService.transferAllMessages(TransferMessagesRequest(dlqSqs, "dlqUrl", queueSqs, "queueUrl"))
+
+        verify(queueSqs).sendMessage("queueUrl", "message-body")
+      }
+
+      @Test
+      fun `Returns the message`() {
+        val result = queueAdminService.transferAllMessages(TransferMessagesRequest(dlqSqs, "dlqUrl", queueSqs, "queueUrl"))
+
+        assertThat(result.messagesFoundCount).isEqualTo(1)
+        assertThat(result.messages)
+          .extracting(Message::getBody, Message::getReceiptHandle)
+          .containsExactly(tuple("message-body", "message-receipt-handle"))
+      }
+    }
+
+    @Nested
+    inner class MultipleMessages {
+      @BeforeEach
+      fun `finds two message on the dlq`() {
+        whenever(dlqSqs.getQueueAttributes(anyString(), eq(listOf("ApproximateNumberOfMessages")))).thenReturn(
+          GetQueueAttributesResult().withAttributes(mapOf("ApproximateNumberOfMessages" to "2"))
+        )
+        whenever(dlqSqs.receiveMessage(any<ReceiveMessageRequest>()))
+          .thenReturn(
+            ReceiveMessageResult().withMessages(
+              Message().withBody("message-1-body").withReceiptHandle("message-1-receipt-handle")
+            )
+          )
+          .thenReturn(
+            ReceiveMessageResult().withMessages(
+              Message().withBody("message-2-body").withReceiptHandle("message-2-receipt-handle")
+            )
+          )
+      }
+
+      @Test
+      fun `Receives message from the dlq`() {
+        queueAdminService.transferAllMessages(TransferMessagesRequest(dlqSqs, "dlqUrl", queueSqs, "queueUrl"))
+
+        verify(dlqSqs, times(2)).receiveMessage(
+          check<ReceiveMessageRequest> {
+            assertThat(it.queueUrl).isEqualTo("dlqUrl")
+            assertThat(it.maxNumberOfMessages).isEqualTo(1)
+          }
+        )
+      }
+
+      @Test
+      fun `Deletes message from the dlq`() {
+        queueAdminService.transferAllMessages(TransferMessagesRequest(dlqSqs, "dlqUrl", queueSqs, "queueUrl"))
+
+        val captor = argumentCaptor<DeleteMessageRequest>()
+        verify(dlqSqs, times(2)).deleteMessage(captor.capture())
+
+        assertThat(captor.firstValue.receiptHandle).isEqualTo("message-1-receipt-handle")
+        assertThat(captor.secondValue.receiptHandle).isEqualTo("message-2-receipt-handle")
+      }
```

Comment:
Using an argument captor to workaround this issue: https://github.com/mockito/mockito-kotlin/issues/288